server/webSocketServer/index.js

```diff
@@ -0,0 +1,33 @@
+const WebSocket = require('ws');
+const userHolder = require('./userHolder');
+const messageCreator = require('./messageCreator');
+
+const startWebSocketServer = (httpServer) => {
+
+    const webSocketServer = new WebSocket.Server({ server: httpServer });
+    const users = userHolder();
+
+    webSocketServer.on('connection', webSocketConnection => {
+        const userId = users.addUser(webSocketConnection);
+        const user = users.getUser(userId);
+
+        webSocketConnection.on('message', rawMessage => {
+            const parsedMessage = JSON.parse(rawMessage);
+            const messageToSend = messageCreator.createMessage(parsedMessage.data, user);
+            users.getAllOpenUsers(userId).forEach(user => user.connection.send(messageToSend));
```

Comment:
I think its better to get all authenticated subscribers so that the controller can controller who to send the event to and not and not index.js

Reply:
for now we need to send to all open users, if there will be groups then we will call users.getGroupUsers, or users.getAuthenticatedUsers, and so on, but for now i don't see what logic do you want to export?

Reply:
I don't think that in the index.js, I need to know what I receive from usersContainer. I need to know what I ask for - authenticated for example.